roboprop_client/views.py

```diff
@@ -189,12 +250,29 @@ def user_settings(request):
 
 def mymodels(request):
     if request.method == "POST":
-        response = utils.upload_file(request.FILES["file"], "models")
+        file = request.FILES["file"]
+        response = utils.upload_file(file, "models")
         if response.status_code == 201:
             messages.success(request, "Model uploaded successfully")
+            model_name = os.path.splitext(file.name)[0]
+            thumbnails = _get_thumbnails([model_name], "models", gallery=False)
+            if all(thumbnail["image"] is not None for thumbnail in thumbnails):
+                base64_thumbnails = list(thumbnail["image"] for thumbnail in thumbnails)
+                tags, categories, parent_categories, colors = _get_suggested_tags(
+                    base64_thumbnails
+                )
+                request.session["model_meta_data"] = {
+                    "name": model_name,
+                    "tags": tags,
+                    "categories": categories,
+                    "parent_categories": parent_categories,
```

Comment:
How do you use category vs parent_category?

Reply:
Currently they are not particularly separated in terms of use. There is occasion where parent_category return additional results to just category, (E.g a cafe table returning Furniture for categories, but also "wood" for parent_category)

But I think it makes sense to combine this into one "categories" and remove duplicates. Thank you

Reply:
categories and parent_categories is just "categories" now. Thanks!

Reply:
Nice